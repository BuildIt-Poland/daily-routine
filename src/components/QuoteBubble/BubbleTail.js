import styled, { keyframes } from 'styled-components';

import media from '../../styles/media';
import BubbleTailArtwork from './BubbleTailArtwork';

const fadeInAnimation = keyframes`
  0% { opacity: 0 }
  100% { opacity: 1 }
`;

const tailAnimation = keyframes`
  0% { transform: translateY(-20px) rotate(15deg) scaleY(0.8) }
  100% { transform: translateX(0) rotate(0) scaleY(1)}
`;

const BubbleTail = styled(BubbleTailArtwork)`
  position: absolute;
  bottom: -3.2rem;
  left: 10%;
  transform: translateY(-20px);
  animation: ${tailAnimation} 400ms cubic-bezier(0.68, -0.55, 0.265, 1.6) forwards;
  transform-origin: 50% 0;

  ${media.phone`
    left: 50%;
  `}

  #top-blob {
    animation: ${fadeInAnimation} 600ms ease-in 400ms forwards;
  }

  #mid-blob {
    animation: ${fadeInAnimation} 600ms ease-in 800ms forwards;
  }

  #bottom-blob {
    animation: ${fadeInAnimation} 600ms ease-in 1200ms forwards;
  }

  ${media.smallPhone`
    height: 30px;
    bottom: -2.3rem
  `};
`;

export default BubbleTail;
