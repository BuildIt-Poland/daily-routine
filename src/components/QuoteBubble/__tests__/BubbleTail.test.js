import React from 'react';
import { create } from 'react-test-renderer';

import BubbleTail from '../BubbleTail';

jest.mock('../BubbleTailArtwork', () => 'BubbleTailArtwork');

describe('COMPONENT - QuoteBubble BubbleTail', () => {
  it('renders correctly', () => {
    const component = create(<BubbleTail />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
