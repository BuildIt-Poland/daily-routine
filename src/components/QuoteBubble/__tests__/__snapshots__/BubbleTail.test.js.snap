// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - QuoteBubble BubbleTail renders correctly 1`] = `
.c0 {
  position: absolute;
  bottom: -3.2rem;
  left: 10%;
  -webkit-transform: translateY(-20px);
  -ms-transform: translateY(-20px);
  transform: translateY(-20px);
  -webkit-animation: oZTux 400ms cubic-bezier(0.68,-0.55,0.265,1.6) forwards;
  animation: oZTux 400ms cubic-bezier(0.68,-0.55,0.265,1.6) forwards;
  -webkit-transform-origin: 50% 0;
  -ms-transform-origin: 50% 0;
  transform-origin: 50% 0;
}

.c0 #top-blob {
  -webkit-animation: gdMPTF 600ms ease-in 400ms forwards;
  animation: gdMPTF 600ms ease-in 400ms forwards;
}

.c0 #mid-blob {
  -webkit-animation: gdMPTF 600ms ease-in 800ms forwards;
  animation: gdMPTF 600ms ease-in 800ms forwards;
}

.c0 #bottom-blob {
  -webkit-animation: gdMPTF 600ms ease-in 1200ms forwards;
  animation: gdMPTF 600ms ease-in 1200ms forwards;
}

@media (max-width:26.25em) {
  .c0 {
    left: 50%;
    height: 30px;
    bottom: -2.3rem;
  }
}

<BubbleTailArtwork
  className="c0"
/>
`;
